executor/analyze_global_stats.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor
+
+import (
+	"context"
+
+	"github.com/pingcap/tidb/domain"
+	"github.com/pingcap/tidb/infoschema"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/logutil"
+	"go.uber.org/zap"
+)
+
+type globalStatsKey struct {
+	tableID int64
+	indexID int64
+}
+
+type globalStatsInfo struct {
+	isIndex int
+	// When the `isIndex == 0`, histIDs will be the column IDs.
+	// Otherwise, histIDs will only contain the index ID.
+	histIDs      []int64
+	statsVersion int
+}
+
+// globalStatsMap is a map used to store which partition tables and the corresponding indexes need global-level stats.
+// The meaning of key in map is the structure that used to store the tableID and indexID.
+// The meaning of value in map is some additional information needed to build global-level stats.
+type globalStatsMap map[globalStatsKey]globalStatsInfo
+
+func (e *AnalyzeExec) handleGlobalStats(ctx context.Context, needGlobalStats bool, globalStatsMap globalStatsMap) error {
+	statsHandle := domain.GetDomain(e.ctx).StatsHandle()
+	if needGlobalStats {
```

Comment:
Maybe we can early return when `needGlobalStats` is false.

Reply:
In factor, it indeed return early when `needGlobalStats` is false

Reply:
I mean early return in code style.
```
if !needGlobalStats {
    return nil
}
// do things
```
The current is also ok to me.

Reply:
updated